Data/ByteString/Lazy.hs

```diff
@@ -525,6 +526,18 @@ minimum (Chunk c cs) = foldlChunks (\n c' -> n `min` S.minimum c')
                                      (S.minimum c) cs
 {-# INLINE minimum #-}
 
+-- | /O(n)/ 'compareLength' compares the length of a 'ByteString' 
+-- to an 'Int'   
+compareLength :: ByteString -> Int -> Ordering
+compareLength s toCmp = go 0 s
+  where
+    go currLen Empty        = compare currLen toCmp
+    go currLen (Chunk c cs) = let nextLen = currLen + S.length c
+                               in if nextLen > toCmp 
+                                    then GT 
+                                    else go nextLen cs
```

Comment:
Instead of tracking `currLen`, you can subtract `S.length c` from `toCmp` and call `compareLength` recursively on `cs`. This would avoid allocating closure `go`.